app/jobs/ingest_job.rb

```diff
@@ -24,6 +24,7 @@ def perform(model_params:, background_job_result:, signed_ids: {}, globus_ids: {
     background_job_result.try_count += 1
     background_job_result.processing!
     model = Cocina::Models.build_request(model_params.with_indifferent_access)
+    model = GlobusDigestGenerator.generate(cocina: model, globus_ids: globus_ids, validate: false)
```

Comment:
I'm wondering whose responsibility it ought to be to sniff `globus_ids` to determine if Globus digests need to be calculated. Right now, if I read this right, it seems to be the responsibility of `GlobusDigestGenerator#gen_digests`. I do like the idea of the generator owning this responsibility over sniffing all over the place (in this case, only `UpdateJob` and `IngestJob` for now). Could/should the sniff take places sooner in the generator and result in returning the model back more immediately? I could go either way.

Reply:
Yeah, I wanted to avoid duplicate logic in the various places that call it. You're right though, `generate` could return immediately if `globus_ids` is empty.